shared/src/commonMain/kotlin/org/hyperskill/app/debug/presentation/DebugReducer.kt

```diff
@@ -3,59 +3,78 @@ package org.hyperskill.app.debug.presentation
         when (message) {
-            is Message.Initialize -> {
+            is Message.Initialize ->
                 if (state is State.Idle || state is State.Error && message.forceUpdate) {
                     State.Loading to setOf(Action.FetchDebugSettings)
                 } else {
                     null
                 }
-            }
-            is Message.FetchDebugSettingsSuccess -> {
+            is Message.FetchDebugSettingsSuccess ->
                 if (state is State.Loading) {
-                    State.Content(message.debugSettings.endpointConfigType) to emptySet()
+                    State.Content(
+                        currentEndpointConfig = message.debugSettings.endpointConfigType,
+                        selectedEndpointConfig = message.debugSettings.endpointConfigType,
+                        stepNavigationInputText = ""
+                    ) to emptySet()
                 } else {
                     null
                 }
-            }
-            is Message.FetchDebugSettingsFailure -> {
+            is Message.FetchDebugSettingsFailure ->
                 if (state is State.Loading) {
                     State.Error to emptySet()
                 } else {
                     null
                 }
-            }
-            is Message.SelectEndpointConfig -> {
+            is Message.SelectEndpointConfig ->
                 if (state is State.Content) {
-                    state.copy(selectedEndpointConfigType = message.endpointConfigType) to emptySet()
+                    state.copy(selectedEndpointConfig = message.endpointConfig) to emptySet()
                 } else {
                     null
                 }
-            }
-            is Message.ApplySettings -> {
+            // Step navigation
+            is Message.StepNavigationInputTextChanged ->
                 if (state is State.Content) {
-                    state to setOf(Action.UpdateEndpointConfig(state.selectedEndpointConfigType))
+                    state.copy(stepNavigationInputText = message.text) to emptySet()
                 } else {
                     null
                 }
-            }
-            is Message.ApplySettingsSuccess -> {
+            is Message.StepNavigationOpenClicked ->
                 if (state is State.Content) {
-                    state to setOf(Action.ViewAction.RestartApplication)
+                    val stepId = state.stepNavigationInputText.toLongOrNull()
+                    if (stepId != null) {
+                        state to setOf(Action.ViewAction.OpenStep(StepRoute.Learn(stepId)))
+                    } else {
+                        null
+                    }
                 } else {
                     null
                 }
-            }
-            is Message.ApplySettingsFailure -> {
+            // Apply settings
+            is Message.ApplySettingsClicked ->
+                if (state is State.Content) {
```

Comment:
Let's check here again that the message came when the `EndpointConfigType` is not equal to the current one to except unnecessary restart